Update docker deployment documentation (#1817)
FerretDB/FerretDB: README.md

## Quickstart

These steps describe a quick local setup.
They are not suitable for most production use-cases because they keep all data

> inside containers and don't [encrypt incoming connections](https://docs.ferretdb.io/security/#securing-connections-with-tls/).

Reviewer: Please use `../security.md`. Never include a domain name

Author: That's the README file, accessible from github view

Reviewer: Oops! My bad, sorry

   `ferretdb` runs FerretDB.

2. Start services with `docker compose up -d`.

>    To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.

Reviewer 2: ```suggestion
   Fetch the latest version with the command `docker compose pull`.
```
Just less words.

Author: In that case I believe we should put `docker compose pull` before `docker compose up`, as it needs to be ran first. The other thing is that it's optional to run `docker compose pull`

Reviewer 2: Yeah we can do that :+1:

3. If you have `mongosh` installed, just run it to connect to FerretDB.
   If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:

   ```sh

>    docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/ -u username -p password

Reviewer 3: I ran setup step by step until here I get `MongoServerError: Unsupported mechanism 'PLAIN'` error. Does it work for you?

Author: Nice catch! 

I was sure that I've connected with success, but probably I've changed too many things and didn't track the changes. 

Adding the `--authenticationMechanism=PLAIN` flag fix the issue. To ensure that everything works correctly I've run everything on seperate environment :smile:

>    docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/ -u username -p password --authenticationMechanism=PLAIN

Reviewer: I think we should use MongoDB URI everywhere we can: `mongodb://username:password@ferretdb/?authMechanism=PLAIN`

   `ferretdb` runs FerretDB.

2. Start services with `docker compose up -d`.
   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.

3. If you have `mongosh` installed, just run it to connect to FerretDB. It will use credentials passed in `mongosh` flags or MongoDB URI 
   to authenticate to the PostgreSQL database. You'll also need to set `authMechanism` to `PLAIN`. 

>    The example uri would look like: 

Reviewer 4: ```suggestion
   The example URI would look like: 
```

   `ferretdb` runs FerretDB.

2. Start services with `docker compose up -d`.
   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.

3. If you have `mongosh` installed, just run it to connect to FerretDB.
   It will use credentials passed in `mongosh` flags or MongoDB URI to authenticate to the PostgreSQL database.
   You'll also need to set `authMechanism` to `PLAIN`.
   The example URI would look like:

>    ``` sh

Reviewer: ```suggestion
   ```sh
```

Otherwise, it will not render correctly

>    mongodb://username:password@localhost/ferretdb?authMechanism=PLAIN"`

Reviewer: ```suggestion
   mongodb://username:password@localhost/ferretdb?authMechanism=PLAIN
```
